framework/src/main/java/org/checkerframework/common/value/ValueTreeAnnotator.java

```diff
@@ -399,6 +399,18 @@ public Void visitMethodInvocation(MethodInvocationTree tree, AnnotatedTypeMirror
             }
         }
 
+        if (atypeFactory
+                .getMethodIdentifier()
+                .isArrayGetLengthInvocation(tree, atypeFactory.getProcessingEnv())) {
+            List<? extends ExpressionTree> args = tree.getArguments();
+            AnnotatedTypeMirror argType = atypeFactory.getAnnotatedType(args.get(0));
+            AnnotationMirror resultAnno = atypeFactory.createArrayLengthResultAnnotation(argType);
+            if (resultAnno != null) {
+                type.replaceAnnotation(resultAnno);
+            }
+            return null;
+        }
+
```

Comment:
I agree and I tried moving it down but the tests fail. This happens because `methodIsStaticallyExecutable()` return `false` at line 430 for Array.getLength() method. Do you think this is another issue? If so I'll try fixing it as a part of another PR.

Reply:
I agree that `methodIsStaticallyExecutable()` should return true for `Array.getLength()`.  Thank you for reporting the problem.
I believe I have [fixed it](https://github.com/typetools/jdk/commit/639f02202fd2ad44401c858d1300a87a327c2fe4) for you so that you are not blocked.